portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -53,6 +58,7 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
     // Get the input from the form
     String text = getParameterWithDefault(request, "comment", "");
+    numComment = Integer.parseInt(request.getParameter("numComment"));
```

Comment:
I guess the crash comes from here?
parseInt throws exception when passing wrong format string.
why not using getParameterWithDefault? ```getParameterWithDefault(request, "numComment", "0")```

Reply:
This will not compile since the value returned by input type "number" is an int, which will cause crash in getParameterWithDefault().

Reply:
Per https://docs.oracle.com/javaee/7/api/javax/servlet/ServletRequest.html#getParameter-java.lang.String-, request.getParameter always returns String.
Even if it returns int, you could make the function template.